### PR TITLE
Epheph/ch11769/enable ability to search all markets even

### PR DESCRIPTION
**Pull request by bthaile**
https://app.clubhouse.io/augur/story/11769/enable-ability-to-search-all-markets-even-if-they-are-not-already-loaded

### CODE REVIEW

File: src/blockchain/log-processors/market-created.ts [hunk 1]
```diff
@@ -100,6 +105,9 @@ export function processMarketCreatedLog(db: Knex, augur: Augur, log: FormattedEv
             (next: AsyncCallback): void => {
               db.insert(marketsDataToInsert).into("markets").asCallback(next);
             },
+            (next: AsyncCallback): void => {
+              db.raw("insert into search_en(marketId, content) values( ?, ? )", [fullTextStringInsert.marketId, fullTextStringInsert.content]).asCallback(next);
```
Comment (epheph) [inline src/blockchain/log-processors/market-created.ts]: Does the normal db.insert({}) method not work for that virtual table?

Comment (bthaile) [reply src/blockchain/log-processors/market-created.ts]: I’d need to try that. I went with raw b/c it worked

File: src/utils/content-search-builder.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  if (marketsDataToInsert === null) throw new Error("market insert row object can't be null");
+
+  const content = [
+    marketsDataToInsert.marketId,
+    marketsDataToInsert.category,
+    marketsDataToInsert.shortDescription !== null ? marketsDataToInsert.shortDescription : "",
+    marketsDataToInsert.tag1 !== null ? marketsDataToInsert.tag1 : "",
+    marketsDataToInsert.tag2 !== null ? marketsDataToInsert.tag2 : "",
+    marketsDataToInsert.resolutionSource !== null ? marketsDataToInsert.resolutionSource : "",
+    marketsDataToInsert.scalarDenomination != null ? marketsDataToInsert.scalarDenomination : "",
```
Comment (epheph) [inline src/utils/content-search-builder.ts]: I wonder if outcome names would be interesting to add? could be a future improvement

File: src/server/getters/get-markets-search.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import * as Knex from "knex";
+import { Address, MarketsContractAddressRow } from "../../types";
+import { queryModifier, getMarketsWithReportingState } from "./database";
+
+// Returning marketIds should likely be more generalized, since it is a single line change for most getters (awaiting reporting, by user, etc)
+export function getMarketsSearch(db: Knex, universe: Address, search: string, sortBy: string|null|undefined, isSortDescending: boolean|null|undefined, limit: number|null|undefined, offset: number|null|undefined, callback: (err?: Error|null, result?: any) => void): void {
```
Comment (epheph) [inline src/server/getters/get-markets-search.ts]: Can we add a type for `result?: any`?

Comment (bthaile) [reply src/server/getters/get-markets-search.ts]: I agree, for future expansion.

File: src/utils/content-search-builder.ts (hunk 2 above)
Comment (bthaile) [reply src/utils/content-search-builder.ts]: I think only categorical markets use description, I think we should be good.

File: src/server/getters/get-markets-search.ts (hunk 3 above)
Comment (bthaile) [reply src/server/getters/get-markets-search.ts]: changed to return array of strings.

**Pull request merged**